airflow/providers/amazon/aws/operators/s3_bucket.py

```diff
@@ -0,0 +1,119 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains AWS S3 operators.
+"""
+from typing import Optional
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+
+class S3CreateBucketOperator(BaseOperator):
+    """
+    This operator creates an S3 bucket
+
+    :param bucket_name: This is bucket name you want to create
+    :type bucket_name: Optional[str]
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is None or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then default boto3 configuration would be used (and must be
+        maintained on each worker node).
+    :type aws_conn_id: Optional[str]
+    :param region_name: AWS region_name. If not specified fetched from connection.
+    :type region_name: Optional[str]
+    """
+    def __init__(self,
+                 bucket_name,
+                 aws_conn_id: Optional[str] = "aws_default",
+                 region_name: Optional[str] = None,
+                 *args,
+                 **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.bucket_name = bucket_name
+        self.region_name = region_name
+        self.s3_hook = S3Hook(aws_conn_id=aws_conn_id, region_name=region_name)
+
+    def execute(self, context):
+        S3CreateBucketOperator.check_for_bucket(self.s3_hook, self.bucket_name)
+        self.s3_hook.create_bucket(bucket_name=self.bucket_name, region_name=self.region_name)
+        self.log.info("Created bucket with name: %s", self.bucket_name)
+
+    @staticmethod
+    def check_for_bucket(s3_hook: S3Hook, bucket_name: str) -> None:
+        """
+        Override this method if you don't want to raise excaption if bucket already exists.
+
+        :param s3_hook: Hook to interact with aws s3 services
+        :type s3_hook: S3Hook
+        :param bucket_name: Bucket name
+        :type bucket_name: str
+        :return: None
+        :rtype: None
+        """
+        if s3_hook.check_for_bucket(bucket_name):
+            raise AirflowException(f"The bucket name {bucket_name} already exists")
```

Comment:
I would suggest to add an arg to the ctor `check_if_exists=True` - not have a method to override.
```python
        if check_if_exists and s3_hook.check_for_bucket(bucket_name):
            raise AirflowException(f"The bucket name {bucket_name} already exists")
```
WDYT?

Reply:
The reason I did it this way was cus if someone wants to perform extra operations post the check like for eg, log some data / fire an email etc they can do so! Not sure if this would cause any issues??

Reply:
If someone wants to do extra operation he/she can do so by having a task (operator) calling before or after this one, don't you think?

Reply:
Makes sense!

Reply:
Actually, sorry but I think it is not even necessary to put behind a flag like `check_if_...`. Can you revert it but also not have a method to override? The check should always happen in my opinion and raising an exception is the correct behavior.